base/linalg/cholesky.jl

```diff
@@ -84,14 +84,17 @@ function chol!{T}(A::AbstractMatrix{T}, uplo::Symbol)
     return uplo == :U ? UpperTriangular(A) : LowerTriangular(A)
 end
 
-function cholfact!{T<:BlasFloat}(A::StridedMatrix{T}, uplo::Symbol=:U; pivot=false, tol=0.0)
+cholfact!{T<:BlasFloat}(A::StridedMatrix{T}, uplo::Symbol=:U, pivot::Union(Type{Val{false}}, Type{Val{true}})=Val{false}; tol=0.0) =
+    _cholfact!(A, pivot, uplo, tol=tol)
+function _cholfact!{T<:BlasFloat}(A::StridedMatrix{T}, ::Type{Val{false}}, uplo::Symbol=:U; tol=0.0)
```

Comment:
This would require changing the API as dispatching on `Val{false}` and `Val{true}` would require `pivot` to not be a default argument and hence change places with `uplo`. It would still require 3 functions, and since `_cholfact` are not exported anyway I preferred to not changed the API

What do you think?